static/src/javascripts/projects/common/modules/commercial/prebid/prebid-service.js

```diff
@@ -0,0 +1,130 @@
+                    window.pbjs.que.push(function () {
+                        pbjs.addAdUnits(pbjsAdUnits);
+                        pbjs.requestBids({
+                            bidsBackHandler : function () {
+                                // We are not actually going to use the bid responses; this test purely validates performance
+                                googletag.display(slotId);
+                                resolve();
+                            }
+                        });
+                    });
+                });
+            });
+
+            return pendingRequests;
+        };
+
+    }
+
+    function PrebidAdUnit(slotElementId) {
+        this.code = slotElementId;
+        this.sizes = getSlotAdSizes(slotElementId);
+        this.bids = getAppnexusBids(this.sizes);
+    }
+
+    function getAppnexusBids(slotSizes) {
+        var slotPlacements = placements;
+
+        slotPlacements = filter(slotPlacements, function matchEdition(placement) {
+            return placement.edition === config.page.edition;
+        });
+
+        slotPlacements = filter(slotPlacements, function matchViewport(placement) {
+            return detect.isBreakpoint(placement.breakpoint);
+        });
+
+        slotPlacements = filter(slotPlacements, function matchSlotSize(placement) {
+            return find(slotSizes, function (slotSize) {
+                return slotSize[0] === placement.width && slotSize[1] === placement.height;
+            });
+        });
+
+        return map(slotPlacements, function toPrebidPlacement(placement) {
+            return {
+                bidder : 'appnexus',
+                params : {
+                    placementId : placement.id,
+                    referrer : 'http://www.theguardian.com/uk'
+                }
+            };
+        });
+    }
+
+    function getSlotAdSizes(slotElementId) {
+        var $slot = bonzo(document.getElementById(slotElementId));
+
+        var validBreakpoints = filter(detect.breakpoints, function (breakpoint) {
+            return detect.isBreakpoint({min : breakpoint.name});
+        });
+
+        var validBreakpointKeys = map(validBreakpoints, function (breakpoint) {
+            // Duplicated from dfp-api for the purpose of our test code
+            return breakpoint.name.replace(/([a-z])([A-Z])/g, '$1-$2').toLowerCase();
+        });
+
+        var largestValidBreakpointDefinition = findLast(validBreakpointKeys, function (key) {
+            return $slot.data(key) !== undefined;
+        });
+
+        var sizeMappingString = $slot.data(largestValidBreakpointDefinition);
+        var sizeStrings = sizeMappingString.split('|');
+        return map(sizeStrings, function (sizeString) {
+            var dimensions = sizeString.split(',');
+            return map(dimensions, function (widthOrHeightString) {
+                return parseInt(widthOrHeightString, 10);
+            });
+        });
+    }
+
+    return new PrebidTestService();
+
+});
```

Comment:
This is clean :-)

Do you need classes (constructors) at all? You're deliberately avoiding keeping state on the class object, which is great, but it's one small step for someone to change that. (See `save-for-later.js`.)

Could this instead be done just using objects and functions? E.g.

``` js
var prebidTestService = {
  loadDependencies: function () {
    require(['js!prebid.js']);
  }
}
```

Or even just:

``` js
var loadPreBidTestDependencies = function () {
  require(['js!prebid.js']);
}
```

I think this approach might be safer in terms of future maintainability, but it also uses less features of the language, which I think makes this simpler.